R/FilterState-utils.R

```diff
@@ -78,213 +88,294 @@ init_filter_state <- function(x,
+                                      varlabel = character(0),
+                                      keep_na = NULL,
+                                      keep_inf = NULL,
+                                      fixed = FALSE,
                                       extract_type = character(0)) {
   if (is.null(varlabel)) varlabel <- character(0)
-  LogicalFilterState$new(
+
+  args <- list(
     x = x,
     x_reactive = x_reactive,
+    dataname = dataname,
     varname = varname,
+    choices = choices,
+    selected = selected,
     varlabel = varlabel,
-    dataname = dataname,
+    keep_na = keep_na,
+    keep_inf = keep_inf,
+    fixed = fixed,
     extract_type = extract_type
   )
+
+  do.call(LogicalFilterState$new, args)
 }
 
 #' @keywords internal
 #' @export
 init_filter_state.numeric <- function(x,
                                       x_reactive = reactive(NULL),
-                                      varname,
-                                      varlabel = attr(x, "label"),
-                                      dataname = NULL,
+                                      dataname,
+                                      varname = attr(x, "label"),
+                                      choices = NULL,
+                                      selected = NULL,
+                                      varlabel = character(0),
+                                      keep_na = NULL,
+                                      keep_inf = NULL,
+                                      fixed = FALSE,
                                       extract_type = character(0)) {
   if (is.null(varlabel)) varlabel <- character(0)
+
+  args <- list(
+    x = x,
+    x_reactive = x_reactive,
+    dataname = dataname,
+    varname = varname,
+    choices = choices,
+    selected = selected,
+    varlabel = varlabel,
+    keep_na = keep_na,
+    keep_inf = keep_inf,
+    fixed = fixed,
+    extract_type = extract_type
+  )
+
   if (length(unique(x[!is.na(x)])) < getOption("teal.threshold_slider_vs_checkboxgroup")) {
-    ChoicesFilterState$new(
-      x = x,
-      x_reactive = x_reactive,
-      varname = varname,
-      varlabel = varlabel,
-      dataname = dataname,
-      extract_type = extract_type
-    )
+    do.call(ChoicesFilterState$new, args)
   } else {
-    RangeFilterState$new(
-      x = x,
-      x_reactive = x_reactive,
-      varname = varname,
-      varlabel = varlabel,
-      dataname = dataname,
-      extract_type = extract_type
-    )
+    do.call(RangeFilterState$new, args)
   }
 }
 
 #' @keywords internal
 #' @export
 init_filter_state.factor <- function(x,
                                      x_reactive = reactive(NULL),
-                                     varname,
-                                     varlabel = attr(x, "label"),
-                                     dataname = NULL,
+                                     dataname,
+                                     varname = attr(x, "label"),
```

Comment:
label is not a varname, probably `deparse(substitute(x))` is more accurate
```suggestion
                                     varname,
```